src/python/nxt_python.c

```diff
@@ -677,7 +707,14 @@ nxt_python_atexit(void)
 
     if (nxt_py_targets != NULL) {
         for (i = 0; i < nxt_py_targets->count; i++) {
-            Py_XDECREF(nxt_py_targets->target[i].application);
+            target = &nxt_py_targets->target[i];
+
+            Py_XDECREF(target->application);
+            Py_XDECREF(target->py_prefix);
+
+            if (target->prefix.length > 0) {
```

Comment:
Do we really need this check?  What does the string contain when the length is 0?  I'd expect a NULL pointer, which can be passed to *free(3)* safely.  Or is it maybe an uninitialized pointer?  Please check.

Reply:
After consulting Stack Overflow, it seems the result of calling `malloc` with a size of 0 will return either `NULL` or a pointer that should never be dereferenced, depending on the implementation. Either way, the result is always safe to pass to `free`, so I removed the check.